sale_order_project/wizard/sale_order_createproject.py

```diff
@@ -0,0 +1,46 @@
+from datetime import date
+from odoo import api, fields, models, _
+from odoo.exceptions import Warning
+
+
+class SaleOrderCreateProject(models.TransientModel):
+    """ wizard to create a Project from a Sale Order """
+    _name = "sales.order.createproject"
+
+    @api.model
+    def default_get(self, fields):
+        result = super(SaleOrderCreateProject, self).default_get(fields)
+        sale_order_id = self.env.context.get('active_id')
+        if sale_order_id:
+            result['sale_order_id'] = sale_order_id
+        return result
+
+    sale_order_id = fields.Many2one('crm.lead', string='Lead', domain=[('type', '=', 'lead')])
+    related_project_id = fields.Many2one('project.project', string='Project')
+
+    @api.model
+    def _prepare_project_vals(self, order):
```

Comment:
As I said in the other comment, is better if you leave this behaviour in sale.order so you don't have to worry about passing the order. The wizard should only call methods on sale.order or write over the account_analytic_id field to assign an already created project.

Reply:
So, what must i to put in the `wizard` file?

Reply:
The wizard python file must get the input of the user and then do one of two things:
1. Either create a new project calling the method of the sale order "action_create_project".
2. Or write the sale order and create the relation sale order - project.

Reply:
How can i know if the `project` field is empty at clicking in the `Accept` button?

Reply:
I just was looking for but i cannot do it, one moment.

Reply:
```python
class SaleOrderCreateProject(models.TransientModel):
    """ wizard to create a Project from a Sale Order """
    _name = "sale.order.createproject"

    @api.model
    def default_get(self, fields):
        result = super(SaleOrderCreateProject, self).default_get(fields)
        sale_order_id = self.env.context.get('active_id')
        if sale_order_id:
            result['sale_order_id'] = sale_order_id
        return result

    sale_order_id = fields.Many2one('sale.order', string='Order', domain=[('type', '=', 'order')])
    related_project_id = fields.Many2one('project.project', string='Project')

    @api.multi
    def action_create_project_task(self):
        self.ensure_one()
        # get the lead to transform
        order = self.sale_order_id
        project = self.related_project_id

        if order.related_project_id:
        # if related_project_id is empty
            # create new project.project
            return order.action_create_project()
        else:
        # else
            # update sale.order.related_project_id with the selected project.project.id
            vals = {
                'related_project_id': self.related_project_id,
            }
            order.write(vals)
            return True
```

Reply:
It is not checking if the field is empty or not, i choose a project and it try to do `rder.action_create_project()`, so i get this error:
```bash
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 935, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 923, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 685, in call_kw
    method = getattr(type(model), name)
AttributeError: type object 'sale.order.createproject' has no attribute 'action_create_project'
```

Reply:
Because your ```order``` is an instance of sale.order.createproject instead of sale.order.

Reply:
It means i have to get a `sale.order` with the current `order.id` and then call the action, right?

Reply:
Done!